src/credential.rs

```diff
@@ -201,11 +206,30 @@ impl CredentialFlat {
         }
 
         if let Some(pass) = credential.password_safe {
-            cred.login = Self::get_bytes_or_none_if_empty(pass.login)?;
-            cred.password = Self::get_bytes_or_none_if_empty(pass.password)?;
-            cred.metadata = Self::get_bytes_or_none_if_empty(pass.metadata)?;
+            cred.login = Self::get_bytes_if_not_empty_or_none(pass.login)?;
+            cred.password = Self::get_bytes_if_not_empty_or_none(pass.password)?;
+            cred.metadata = Self::get_bytes_if_not_empty_or_none(pass.metadata)?;
         }
 
         Ok(cred)
     }
+
+    /// Update credential fields with new values, and save
+    pub fn update_from(&mut self, update_req: CredentialUpdate) -> Result<(), Status> {
+        if let Some(new_label) = update_req.new_label {
+            self.label = ShortData::from_slice(new_label).map_err(|_| Status::NotEnoughMemory)?;
+        }
+        if let Some(p) = update_req.properties {
+            self.touch_required = p.touch_required();
+        }
+        if let Some(pws) = update_req.password_safe {
+            self.login = Self::get_bytes_if_not_empty_or_none(pws.login)
+                .map_err(|_| Status::UnspecifiedNonpersistentExecutionError)?;
+            self.password = Self::get_bytes_if_not_empty_or_none(pws.password)
+                .map_err(|_| Status::UnspecifiedNonpersistentExecutionError)?;
+            self.metadata = Self::get_bytes_if_not_empty_or_none(pws.metadata)
+                .map_err(|_| Status::UnspecifiedNonpersistentExecutionError)?;
```

Comment:
Shouldn’t we ignore `None` values here (instead of returning an error)?

Reply:
Error is only on conversion issue, not if None is encountered. `.map()` would be ideal here, but I had problem with dealing with the said error in the `Fn`.